src/wasm/wasm.cpp

```diff
@@ -217,15 +217,8 @@ const char* getExpressionName(Expression* curr) {
 }
 
 Literal getSingleLiteralFromConstExpression(Expression* curr) {
-  if (auto* c = curr->dynCast<Const>()) {
-    return c->value;
-  } else if (auto* n = curr->dynCast<RefNull>()) {
-    return Literal::makeNull(n->type);
-  } else if (auto* r = curr->dynCast<RefFunc>()) {
-    return Literal::makeFunc(r->func);
-  } else {
-    WASM_UNREACHABLE("Not a constant expression");
-  }
+  assert(Properties::isConstantExpression(curr));
+  return Properties::getSingleLiteral(curr);
```

Comment:
Huh, it doesn't seem useful to have two functions that do the same thing. It would be good to add a TODO here to remove one of them.

Reply:
This function's body was a duplicate of `Properties::getSingleLiteral(curr)`, yeah, but I added an additional assert here to make sure that it is only used where `Properties::isConstantExpression(curr)` is true, in case `Properties::getSingleLiteral` ever diverges to support more than what's valid in global initializers. Might well be that it never does, and is actually intended to do the exact same thing anyway?